lettuce/ext/_reporter/nan_repoter.py

```diff
@@ -0,0 +1,196 @@
+import os
+from typing import List
+
+import torch
+import numpy as np
+
+from ... import Reporter, Simulation, BreakableSimulation
+from .vtk_reporter import VTKReporter
+from timeit import default_timer as timer
+
+__all__ = ["NaNReporter", "HighMaReporter"]
+
+
+class NaNReporter(Reporter):
+    """reports any NaN and aborts the simulation"""
+    # WARNING: too many NaNs in very large simulations can confuse torch and
+    # trigger an error, when trying to create and store the nan_location
+    # tensor.
+    # ...to avoid this, leave outdir=None to omit creation and file-output of
+    # nan_location. This will not impact the abortion of sim. by NaN_Reporter
+
+    def __init__(self, interval=100, outdir=None, vtk=False):
+        self.outdir = outdir
+        self.vtk = vtk
+        self.name = 'NaN'  # what is the name for?
+        self.failed_iteration = None
+        super().__init__(interval)
+
+    def __call__(self, simulation: 'Simulation'):
```

Comment:
Since you import `Simulation` with no apparent issues, you can type it directly. Keep in mind, though: This must be a `BreakableSimulation`

Reply:
@MaxBille have you tried this?